sdk/tables/azure-data-tables/inc/azure/data/tables/table_client.hpp

```diff
@@ -128,21 +128,29 @@ namespace Azure { namespace Data { namespace Tables {
      */
     Response<Models::DeleteEntityResult> DeleteEntity(
         Models::TableEntity const& tableEntity,
-        Core::Context const& context = {});
+        Core::Context const& context = {}) const;
 
     /**
-     * @brief Upsert specified entity in a table.
+     * @brief Update or insert specified entity in a table.
      *
-     * @param tableEntity The TableEntity to upsert.
-     * @param upsertKind Upsert kind(update/merge), default update.
+     * @param tableEntity The TableEntity to update or insert.
      * @param context for canceling long running operations.
      * @return Upsert entity result.
      */
-    Response<Models::UpsertEntityResult> UpsertEntity(
+    Response<Models::UpdateEntityResult> UpdateInsertEntity(
         Models::TableEntity const& tableEntity,
-        Models::UpsertKind upsertKind = Models::UpsertKind::Update,
-        Core::Context const& context = {});
+        Core::Context const& context = {}) const;
 
+    /**
+     * @brief Merge or insert the specified entity in a table.
+     *
+     * @param tableEntity The TableEntity to merge or insert.
+     * @param context for canceling long running operations.
+     * @return Upsert entity result.
+     */
+    Response<Models::MergeEntityResult> MergeInsertEntity(
```

Comment:
How does the caller distinguish between Merge and Insert entity?

The implementation just forwards to MergeEntity, so maybe the question is: Is MergeEntity *always* supposed to create an entity if it does not exist?

If so, how do I perform a MergeEntity that fails if the item does not exist?

Reply:
done based on call